tasks/codex/edm.py

```diff
@@ -59,28 +75,123 @@ def transfer_data_encounter():
             else:  # TODO handle better?
                 raise ValueError(f'unknown locationId "{loc}" on enc {enc.guid}')
 
-        # FIXME setting customField values should also be some kind of setter on the objects that does
-        #   this check instead (and checks type, values [against choices] etc.
-        edm_custom_fields = edm_data.get('customFields')
-        if edm_custom_fields:
-            custom_fields = {}
-            for cfd_id in edm_custom_fields:
-                defn = SiteSettingCustomFields.get_definition('Encounter', cfd_id)
-                if not defn:  # TODO handle better?
-                    raise ValueError(
-                        f'invalid customField definition id {cfd_id} on enc {enc.guid}: {edm_custom_fields}'
-                    )
-                custom_fields[cfd_id] = edm_custom_fields[cfd_id]
-            enc.custom_fields = custom_fields
+        tx_id = edm_data.get('taxonomy')
+        if tx_id:
+            Taxonomy(tx_id)  # will raise ValueError if bad id
+            enc.taxonomy_guid = tx_id
+
+        edm_custom_fields = edm_data.get('customFields', {})
+        for cfd_id in edm_custom_fields:
+            enc.set_custom_field_value(cfd_id, edm_custom_fields[cfd_id])
     print('encounter edm data transfer complete')
 
 
 def transfer_data_sighting():
-    pass
+    import app.modules.utils as util
+    from app.modules.sightings.models import Sighting
+    from app.modules.site_settings.models import Regions, Taxonomy
+
+    reg = Regions()
+    sightings = Sighting.query.all()
+    print('sighting edm data transfer started')
+    for sighting in tqdm.tqdm(sightings):
```

Comment:
I've propbably missed this but how do we know that every sighting on EDM has an equivalent sighing in Houston and what is this tqdm interface anyway? Why are you not reading all the sightings from the DB

Reply:
1. the Migration is what builds the houston sightings.  there _might_ (read: usually probably are) a few **less** sightings in houston (due to _very_ wonky data that doesnt migrate), so using houston to iterate through should be safe
2. tqdm is just the little progress bar thing that (somehow, magically) happens.  from a functional perspective, this acts exactly as if it was `for sighting in sightings:`  :exploding_head: